src/daemon/meson.build

```diff
@@ -3,24 +3,30 @@ daemon_deps = [
     giomm_dep,
     glib_dep,
     glibmm_dep,
-    tdbs_dbus_dep
+    uim_dbus_dep
 ]
 
 daemon_sources = [
     'arguments.cpp',
     'arguments.h',
+    'configuration.cpp',
+    'configuration.h',
     'daemon.cpp',
     'daemon.h',
     'dbus_service.cpp',
-    'dbus_service.h'
+    'dbus_service.h',
+    'id_source.cpp',
+    'id_source.h',
+    'id_sources/mass_storage_device_id_source.cpp',
```

Comment:
Should be possible to configure at build time which sources to make available

Reply:
The MSD id source will effectively not be included in the binary if the msd_id_source build option is set to false. MSD source will never be instantiated due to this in `src/daemon/id_source.cpp:create_sources()`:

```c++
#if UIM_CONFIG_MASS_STORAGE_DEVICE_ID_SOURCE
            sources.emplace_back(std::make_unique<MassStorageDeviceIdSource>());
#endif
```

There is no harm in building this file unconditionally though since no extra dependencies are needed and by always building it it will always be checked etc. But, perhaps we should complicate the build files anyway, even if it is not needed, to set an example for other sources.

Reply:
Sorry, I missed the meson build option file.
For now, it might be enough. Not sure if there is a need in the future with a more complicated build in case of customized hardware etc.

Reply:
With other sources that requires libraries it might make sense to leave parts of that in too and mock some stuff so other parts can be (unit) tested. So I think I am leaning towards leaving this as is... and see how the first "real" source turns out. Might be different things that makes sense for different "real" sources as well... Important thing is that nothing related to a source ends up in the daemon binary if it is disabled.

Reply:
Should of course not have large source files with a bunch of #if:ery, but breaking larger sources out into components might make sense. And making the components testable even if they are not included in the final executable... meh, I hope you have managed to convey what I am thinking.